chore: Update changelog for v1.1.5 (#381)
awslabs/aws-mysql-jdbc: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [1.1.5] - 2023-03-29
### Changed

> * Improve clusterHostInstancePattern example in README.md.

Reviewer: ```suggestion
* Improved `clusterHostInstancePattern` example in README.md.
```

Reviewer: More of a general comment regarding the documentation changes, are we also able to link to the affected section here as a hyperlink?

Author: Done!

Author: Yup! Should be able to. Only thing is that it won't highlight the diffs.

Author: Done! The Maven Central URL changes had multiple updates, so I linked each one.

> * Optimize thread locks and expiring cache for Enhanced Monitoring Plugin. ([PR #356](https://github.com/awslabs/aws-mysql-jdbc/pull/356))

Reviewer: ```suggestion
* Optimized thread locks and expiring cache for the Enhanced Monitoring Plugin. ([PR #356](https://github.com/awslabs/aws-mysql-jdbc/pull/356))
```

Author: Done!

* Check log level before logTrace and logDebug with Messages.getString for better log consistency. ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367))

> * Update README.md with the latest Maven Central URLs.

Reviewer: ```suggestion
* Updated README.md with the latest Maven Central URLs.
```

Author: Done!

> * Refactor reduce map lookups to reduce number of locks required. ([PR #373](https://github.com/awslabs/aws-mysql-jdbc/pull/373))

Reviewer: ```suggestion
* Refactored reduce map lookups to reduce number of locks required. ([PR #373](https://github.com/awslabs/aws-mysql-jdbc/pull/373))
```

Author: Done!

* Updated README.md to reflect that the driver does not support custom RDS clusters.

> * Update DBeaver documentation in README.md with instructions on how to enable database selectors in the UI.

Reviewer: ```suggestion
* Updated DBeaver documentation in README.md with instructions on how to enable database selectors in the UI.
```

Author: Done!

> * Update README.md to reflect that Amazon RDS Blue/Green Deployments are not supported.

Reviewer: ```suggestion
* Updated README.md to reflect that Amazon RDS Blue/Green Deployments are not supported.
```

Author: Done!

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

## [1.1.5] - 2023-03-29
### Changed

> * Optimized thread locks and expiring cache for the Enhanced Monitoring Plugin. ([PR #356](https://github.com/awslabs/aws-mysql-jdbc/pull/356)) ([PR #373](https://github.com/awslabs/aws-mysql-jdbc/pull/373))

Author: Done something very similar

> * Added checking log level before logTrace and logDebug with Messages.getString for better log consistency. ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367))

Reviewer 2: Check log level to avoid unnecessary resource allocation

Author: Done!

> * Check log level to avoid unnecessary resource allocation ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367)).

Reviewer 2: ```suggestion
* Checked log level to avoid unnecessary resource allocation ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367)).
```

Author: Done!

> * Refactored reduce map lookups to reduce number of locks required. ([PR #373](https://github.com/awslabs/aws-mysql-jdbc/pull/373))

Reviewer 2: You can add this PR to the first point `Optimized thread locks...`

Author: Checked in directly with Karen and we can ignore this one.

>   * `clusterInstanceHostPattern` Failover Parameter example ([here](https://github.com/awslabs/aws-mysql-jdbc#failover-parameters)).

Reviewer 2: ```suggestion
  * `clusterInstanceHostPattern` failover parameter example ([here](https://github.com/awslabs/aws-mysql-jdbc#failover-parameters)).
```

Author: Done!

> * Updated README.md with improvements to the `clusterHostInstancePattern` parameter example.

Reviewer: Thanks, since we have a number of documentation changes, do you think this would help clean up the text? In addition, are we able to hyperlink to the exact section header to where the change occurred?

* Updated README.md with the following:
  * Made improvements to the `clusterHostInstancePattern` parameter example.
  * Latest Maven Central URLs.
  * Clarified that the driver does not support custom RDS clusters.
  * Updated DBeaver documentation with instructions on how to enable database selectors in the UI.
  * Clarified that Amazon RDS Blue/Green Deployments are not supported.

Author: Love it! Gonna do that

Author: Done!

>   * Latest Maven Central URLs.([Link #1](https://github.com/awslabs/aws-mysql-jdbc#direct-download-of-the-jar-file), [Link #2](https://github.com/awslabs/aws-mysql-jdbc#as-a-maven-dependency), [Link #3](https://github.com/awslabs/aws-mysql-jdbc#as-a-gradle-dependency)) 

Reviewer 3: Would it make sense to change this to the following to avoid mystery Links with #'s:

 * Latest Maven Central URLs for [direct download of the jar file](https://github.com/awslabs/aws-mysql-jdbc#direct-download-of-the-jar-file), [as a maven dependency](https://github.com/awslabs/aws-mysql-jdbc#as-a-maven-dependency), and [as a gradle dependency](https://github.com/awslabs/aws-mysql-jdbc#as-a-gradle-dependency)

Author: done

* Updated README.md to reflect that the driver does not support custom RDS clusters.
* Updated DBeaver documentation in README.md with instructions on how to enable database selectors in the UI.

> * Updated README.md to reflect that Amazon RDS Blue/Green Deployments are not supported.

Reviewer 2: Can we group the readme changes for:
1. unsupported items
2. updates/improvements to instructions

Author: Done!

>   * Updated DBeaver documentation with instructions on how to enable database selectors in the UI. ([Link](https://github.com/awslabs/aws-mysql-jdbc#using-the-driver-with-a-database-client-dbeaver))

Reviewer 3: could also probably remove words like "Updated" and maybe "Improvements" since you already indicated these are doc updates in your first line

>   * Updated DBeaver documentation with instructions on how to enable database selectors in the UI. ([Link](https://github.com/awslabs/aws-mysql-jdbc#using-the-driver-with-a-database-client-dbeaver))

Author: Done

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).

> ## [1.1.5] - 2023-03-29

Reviewer 2: I'd change this to march 31

Author: Done

### Changed

* Optimized thread locks and expiring cache for the Enhanced Monitoring Plugin. ([PR #356](https://github.com/awslabs/aws-mysql-jdbc/pull/356))

> * Added checking log level to avoid unnecessary resource allocation. ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367))

Reviewer 2: ```suggestion
* Check log level to avoid unnecessary resource allocation. ([PR #367](https://github.com/awslabs/aws-mysql-jdbc/pull/367))
```

Author: Done

* Updated README.md with the following
  * Improvements to the `clusterInstanceHostPattern` Failover Parameter example. ([Link](https://github.com/awslabs/aws-mysql-jdbc#failover-parameters))
  * Latest Maven Central URLs.([Link #1](https://github.com/awslabs/aws-mysql-jdbc#direct-download-of-the-jar-file), [Link #2](https://github.com/awslabs/aws-mysql-jdbc#as-a-maven-dependency), [Link #3](https://github.com/awslabs/aws-mysql-jdbc#as-a-gradle-dependency)) 
  * Updated DBeaver documentation with instructions on how to enable database selectors in the UI. ([Link](https://github.com/awslabs/aws-mysql-jdbc#using-the-driver-with-a-database-client-dbeaver))
  * Clarified that the driver does not support custom RDS clusters. ([Link](https://github.com/awslabs/aws-mysql-jdbc#connection-url-descriptions))
  * Clarified Amazon RDS Blue/Green Deployments are not supported. ([Link](https://github.com/awslabs/aws-mysql-jdbc#amazon-rds-bluegreen-deployments))

### Fixed
* Only update topology for specific methods so that certain workflows are not interrupted. ([Issue 363](https://github.com/awslabs/aws-mysql-jdbc/issues/363))

> * Fixed methods passing in JdbcConnection proxies by adding checks before casting to ConnectionImpl.  ([Issue 365](https://github.com/awslabs/aws-mysql-jdbc/issues/365))

Reviewer 2: ```suggestion
* Fixed methods passing in JdbcConnection proxies by adding checks before casting to ConnectionImpl. ([Issue 365](https://github.com/awslabs/aws-mysql-jdbc/issues/365))
```

Author: Done
